cli/mender-artifact/write.go

```diff
@@ -323,7 +323,7 @@ func makeUpdates(ctx *cli.Context) (*awriter.Updates, error) {
 
 // makeTypeInfo returns the type-info provides and depends and the augmented
 // type-info provides and depends, or nil.
-func makeTypeInfo(ctx *cli.Context) (*artifact.TypeInfoV3, *artifact.TypeInfoV3, error) {
+func makeTypeInfo(ctx *cli.Context, typ string) (*artifact.TypeInfoV3, *artifact.TypeInfoV3, error) {
```

Comment:
Question: I don't understand, how is this different from `ctx.String("type")` ? That is what is passed in anyways

Reply:
The point is that for `rootfs-image` Artifacts, `type` is not specified by the user and hence defaults to an empty string. For `module-image` it worked because `type` is mandatory.

Reply:
Oh, missed the `writeRootfs` call.

Reply:
Twice!

Reply:
I am also playing with the idea of setting the default value in `cli.Context`, how about it?

Reply:
The flag is not valid for `write rootfs-image`, so it can't have a default value.